samples/npm1300_fuel_gauge/src/fuel_gauge.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/npm1300_charger.h>
+#include <zephyr/sys/printk.h>
+
+#include "nrf_fuel_gauge.h"
+
+static float max_charge_current;
+static float max_discharge_current;
+static int64_t ref_time;
+
+static const struct battery_model battery_model = {
+#include "battery_model.inc"
+};
+
+static void read_sensors(const struct device *charger, float *voltage, float *current, float *temp)
+{
+	struct sensor_value value;
+
+	sensor_sample_fetch(charger);
+
+	sensor_channel_get(charger, SENSOR_CHAN_GAUGE_VOLTAGE, &value);
+	*voltage = (float)value.val1 + ((float)value.val2 / 1000000);
+
+	sensor_channel_get(charger, SENSOR_CHAN_GAUGE_TEMP, &value);
+	*temp = (float)value.val1 + ((float)value.val2 / 1000000);
+
+	sensor_channel_get(charger, SENSOR_CHAN_GAUGE_AVG_CURRENT, &value);
+	*current = (float)value.val1 + ((float)value.val2 / 1000000);
+}
+
+void fuel_gauge_init(const struct device *charger)
+{
+	struct sensor_value value;
+	struct nrf_fuel_gauge_init_parameters parameters = { .model = &battery_model };
+
+	read_sensors(charger, &parameters.v0, &parameters.i0, &parameters.t0);
+
+	/* Store max charge/discharge current, needed for ttf calculation */
+	sensor_channel_get(charger, SENSOR_CHAN_GAUGE_DESIRED_CHARGING_CURRENT, &value);
+	max_charge_current = (float)value.val1 + ((float)value.val2 / 1000000);
+
+	sensor_channel_get(charger, SENSOR_CHAN_GAUGE_MAX_LOAD_CURRENT, &value);
+	max_discharge_current = (float)value.val1 + ((float)value.val2 / 1000000);
+
+	nrf_fuel_gauge_init(&parameters, NULL);
+
+	ref_time = k_uptime_get();
+}
+
+void fuel_gauge_update(const struct device *charger)
+{
+	float voltage;
+	float current;
+	float temp;
+	float soc;
+	float tte;
+	float ttf;
+	float delta;
+
+	read_sensors(charger, &voltage, &current, &temp);
+
+	delta = (float) k_uptime_delta(&ref_time) / 1000.f;
+
+	soc = nrf_fuel_gauge_process(voltage, current, temp, delta, NULL);
+	tte = nrf_fuel_gauge_tte_get();
+	ttf = nrf_fuel_gauge_ttf_get(max_charge_current, max_discharge_current);
```

Comment:
The second parameter to `nrf_fuel_gauge_ttf_get` should be the termination current, e.g. typically 10% of the charge current.